### PR TITLE
Gh 1824 operation field summary

### PR DESCRIPTION
**Pull request by p013570**

### CODE REVIEW

File: core/data/src/main/java/uk/gov/gchq/gaffer/data/element/EdgeDirection.java [hunk 1]
```diff
@@ -16,9 +16,12 @@
 
 package uk.gov.gchq.gaffer.data.element;
 
+import uk.gov.gchq.koryphe.Summary;
+
 /**
  * Enumerated type denoting the directionality of an {@link Edge}.
  */
+@Summary("The directed of an edge")
```
Comment (m55624) [inline core/data/src/main/java/uk/gov/gchq/gaffer/data/element/EdgeDirection.java]: "The direction" I think makes more sense? This summary is also identical almost to the Summary in DirectedType

File: core/data/src/main/java/uk/gov/gchq/gaffer/data/element/id/DirectedType.java [hunk 2]
```diff
@@ -16,10 +16,13 @@
 
 package uk.gov.gchq.gaffer.data.element.id;
 
+import uk.gov.gchq.koryphe.Summary;
+
 /**
  * A {@code DirectedType} defines whether edges should be
  * directed, undirected or either/both.
  */
+@Summary("The direction of Edges")
```
Comment (m55624) [inline core/data/src/main/java/uk/gov/gchq/gaffer/data/element/id/DirectedType.java]: "Is the Edge directed?"

File: rest-api/core-rest/src/main/java/uk/gov/gchq/gaffer/rest/service/v2/OperationServiceV2.java [hunk 3]
```diff
@@ -287,10 +293,14 @@ private Operation generateExampleJson(final Class<? extends Operation> opClass)
             }
 
             if (null != field) {
-                final Required[] annotations = field.getAnnotationsByType(Required.class);
+                required = null != field.getAnnotation(Required.class);
+                summary = getSummaryValue(field.getType());
 
-                if (null != annotations && annotations.length > 0) {
-                    required = true;
+                if (field.getType().isEnum()) {
```
Comment (m55624) [inline rest-api/core-rest/src/main/java/uk/gov/gchq/gaffer/rest/service/v2/OperationServiceV2.java]: There is an if below for the same thing.

**Comment by p013570** [comment]
@m55624 - thanks for making those changes.

**Pull request merged**